### ISSUE TITLE
Importing anchor in a ViteJS(React) project, crashes completely

### ISSUE THREAD

**Issue by version13** [coral-xyz/anchor#986]
Hello,
Using ViteJS and doing anything with the `anchor` ends up in the entire instance crashing and not running.

The error is the following: `global is undefined`

## System:
Windows 10 latest update (21H1 as of writing this for me)
WSL2 Ubuntu fresh install, with only the required tools to run Anchor and React.

## Steps to replicate:
**(Using WSL2)**
1. Create a React project using ViteJS CLI (Vanilla or Typescript)
2. Import anything from `anchor` or anything individual, example `{web3}`
3. Try to do anything with it, in this case let's `console.log(web3)`.
4. ViteJS stops running completely and throws a `global is not defined` error.

## Things tried:
* Switch between `npm`, `pnpm` and `yarn` it fails with all.
* Vanilla and Typescript, it fails with both.
* Remade the project fresh multiple times, cleared npm cache, deleted modules.
* Downgraded anchor twice, still failed.

## What worked
* Switching to CRA (Create React App) throws no error, I've only tried Vanilla right now but will use Typescript soon.
Update: CRA + Typescript works fine too.

**Comment by dimfeld** [comment]
This is https://github.com/near/borsh-js/issues/38

**Comment by dimfeld** [comment]
I opened https://github.com/near/borsh-js/pull/40 to fix this. Note that you will still need a [Buffer polyfill](https://www.npmjs.com/package/buffer) to use Borsh/Anchor in the browser, even after that is merged.

**Comment by mdhrumil** [comment]
While building the project with vite, I got the same error about a month ago. When you solve the global error, you might get `Buffer is not defined ` error.
Here's what I did step-wise and it worked well:

1. npm install buffer

2. Create a file named shims.ts, and add this there:
```
import { Buffer } from 'buffer';
// @ts-ignore
window.Buffer = Buffer;
```

3. Then add this into the script tags of your index.html:
```
<script>
    window.global = window;
 </script>
<script type="module" src="/src/shims.ts"></script>
```

This should solve the global is not defined, but since the buffer is a nodejs module, for bringing it to the browser, you might have to inject it during the build by customizing the vite.config.ts.
Here's how my vite.config.ts looks like:

```
import { defineConfig } from 'vite';
import viteTsconfigPaths from 'vite-tsconfig-paths';
import react from '@vitejs/plugin-react';
import inject from '@rollup/plugin-inject';
import path from 'path';

export default defineConfig({
    plugins: [viteTsconfigPaths(), react()],
    resolve: {
        alias: {
            '@': path.resolve(__dirname, 'src'),
        },
    },
    build: {
        rollupOptions: {
            plugins: [inject({ Buffer: ['buffer', 'Buffer'] })],
        },
    },
});
```

Hopefully, this should solve the error.

**Comment by ilyazub** [comment]
`global is undefined` fix is released in borsh-js v0.7.0: https://github.com/near/borsh-js/issues/38#issuecomment-984606085

**Comment by Henry-E** [comment]
Hopefully this is all sorted now?
